parts/k8s/windowscontainerdfunc.ps1

```diff
@@ -99,10 +99,26 @@ function Select-Windows-Version {
     "18363" { return "1909" }
     "19041" { return "2004" }
     "19042" { return "20H2" }
+    "20348" { return "ltsc2022" }
     Default { return "" } 
   }
 }
 
+# Danger: ReleaseId is deprecated. 
+function Get-WindowsVersion-From-ReleaseId {
+  $windowsReleaseId = (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").ReleaseId
+  $windowsVersion = ""
+
+  # Starting with 20H2 tags used to publish container images may not match the 'ReleaseId'

```

Comment:
It won't work at all for WS 2022 is that right?

Reply:
WS2022 is covered by the buildNumber switch. 
`20348 => ltsc2022`
The `ReleaseId` fallback is for anything that the switch doesn't have a case for.

Reply:
I think our switch covers all known versions BuildId => image tags.  This will cause strange runtime errors for anything newer than 20H2 as you experienced. It maybe be better to fail explicitly here. wdyt?

Reply:
If we could somehow cause aks-engine to hit an exception here, that would be ideal. It would say "Current Build not recognized" or something like that. This way, nobody will be rooting around in logs after-the-fact

Reply:
Are we testing versions that aren't captured in the switch? Would we have to update the corresponding unit tests?

Reply:
unfortunately this is done at cluster bring up time not the cluster template generation.  Even though CSE would fail and some one would have to go get the logs to figure it out, I think that would be ideal over the cluster coming online but having strange behavior when deploying pods and trying to root cause it.  

The reason it needs the version id here instead of just using a multi-arch image is the experimental Hyper-v  support doesn't work with multi-arch images.  You end up in a similar scenario where the cluster is online but pods can't be deployed.